tests/test_pylogger.py

```diff
@@ -0,0 +1,161 @@
+"""Tests for the pylogger module."""
+
+import importlib
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from template_agent.utils.pylogger import _discover_app_loggers
+
+
+class TestDiscoverAppLoggers:
+    """Test cases for _discover_app_loggers function."""
+
+    def setup_method(self):
+        """Reset cache before each test."""
+        # Reset the cache by accessing the module-level variable
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+    def test_discover_loggers_for_real_package(self):
+        """Test discovering loggers for the actual template_agent package."""
+        # Reset cache
+        import template_agent.utils.pylogger as pylogger_module
+
+        pylogger_module._DISCOVERED_LOGGERS_CACHE = None
+
+        # Discover loggers
+        logger_names = _discover_app_loggers("template_agent")
+
+        # Should always include root package
+        assert "template_agent" in logger_names
+
+        # Should find some submodules
+        assert len(logger_names) > 1
+
+        # Should find src and utils subpackages
+        assert any("template_agent.src" in name for name in logger_names)
+        assert any("template_agent.utils" in name for name in logger_names)
```

Comment:
I like the fancy `any()` expressions, but you didn't need them at lines 104-107, so should we be using something simpler here?